packages/grid/x-data-grid/src/components/GridSelectedRowCount.tsx

```diff
@@ -54,12 +54,19 @@ const GridSelectedRowCount = React.forwardRef<HTMLDivElement, GridSelectedRowCou
     const ownerState = useGridRootProps();
     const classes = useUtilityClasses(ownerState);
     const rowSelectedText = apiRef.current.getLocaleText('footerRowSelected')(selectedRowCount);
-
+    const theme: any = useTheme();
+    const themeId = React.useMemo(() => {
+      if (Object.keys(theme).length === 1) {
+        return Object.keys(theme)[0];
```

Comment:
IMO this will work with any variant of the theme object, even that which doesn't comply with the [theme scoping](https://mui.com/material-ui/guides/theme-scoping/) definition (i.e. not using `THEME_ID` from material) but if it's an object with one key, it will be assumed to have theme scoping which doesn't necessarily be true.
Do you propose to replicate this logic everywhere where we have these scoping issues?

Another reason why this isn't a proper solution is, any component wrapped with `styled` from the system shouldn't have access to those theme values which are [injected](https://github.com/mui/material-ui/blob/324a42fab01c7da4348ed70179cf7d29b4531a8f/packages/mui-material/src/styles/createTheme.js#L38-L46) in the `styled` exported from `@mui/material/styles`, so in case we need one of them, we'll have to fallback to the `@mui/material/styles` variant anyways.

Wouldn't it be better to fix this internally in the `@mui/system` so that we don't need to do this workaround everywhere? Tbh it seems a bit hacky to me.
I also left a [comment](https://github.com/mui/mui-x/issues/10484#issuecomment-1743707146) on the issue about the possible solutions I have in my mind.
Let me know what you think about it?